### PR TITLE
ci: add shell-tests job to gates

### PR DESCRIPTION
**Pull request by baksohyeon**
## Problem

On 2026-09-16 during PR #133 gate verification, `.github/workflows/gates.yml` ran pytest and redaction but did not execute any `master-ops/scripts/test-*.sh` shell tests, so promoted shell tests were present yet inert in CI.

## Why this approach

I considered adding shell checks into the existing `tests` job, but a dedicated `shell-tests` matrix job keeps shell-surface failures isolated by OS and gives explicit pass/fail evidence for this promotion step.

## What this changes

- Add a new `shell-tests` job to `.github/workflows/gates.yml` on `ubuntu-latest` and `macos-latest` with `fail-fast: false`.
- Provision Python in `shell-tests` with `actions/setup-python@v7.0.0` and `python-version: "3.12"`, matching the `tests` job.
- Run sorted `master-ops/scripts/test-*.sh`, print `name exit=N` per test, total `passed N failed N skip-lines N`, and emit full output for failing tests.
- Guard zero-match behavior by collecting test paths into an array and failing with an explicit message when no shell tests are present.
- Keep existing `tests` and `redaction` jobs unchanged.

## Expected effect

Reviewers can inspect `shell-tests` on this PR and see both failure-path proof (with the probe) and recovery to success (after probe removal), with per-leg conclusions on Ubuntu and macOS.

## Testing

- [x] `PYTHONPATH=src python3 -m pytest tests -q` — `616 passed, 13 subtests passed in 45.38s`
- [x] A test that fails without this change, or a note on why none was needed — probe commit `42356a7` intentionally added `master-ops/scripts/test-zz-probe.sh` (exit 1), producing shell-tests failures in both legs.
- [x] New files staged before running any scan. The scanners read tracked files, so an unstaged addition is invisible to them and the run comes back green without having looked at it.
- [x] Probe run URL + conclusions (ubuntu/macos): `https://github.com/baksohyeon/mogui-ADE-orchestrator/actions/runs/35335036600` (`shell-tests (ubuntu-latest)=failure`, `shell-tests (macos-latest)=failure`)
- [x] Clean run URL + conclusions (ubuntu/macos): `https://github.com/baksohyeon/mogui-ADE-orchestrator/actions/runs/35335258076` (`shell-tests (ubuntu-latest)=success`, `shell-tests (macos-latest)=success`)
- [x] Post-review run with Python setup step URL + conclusions (ubuntu/macos): `https://github.com/baksohyeon/mogui-ADE-orchestrator/actions/runs/35336729107` (`shell-tests (ubuntu-latest)=success`, `shell-tests (macos-latest)=success`)
- [x] `./scripts/generate-manifest --check` exit code: `0`
- [x] `./scripts/redaction-scan.sh` finding count: `0 findings`; warning emitted: `organization-specific rules not loaded (REDACTION_EXTRA_PATTERNS unset or empty)`
- [x] Local shell-step body output (final HEAD):
  - `test-bash-output-size-warn.sh exit=0`
  - `test-codex-hooks-review-answer.sh exit=0`
  - `test-dispatch-runtime.sh exit=0`
  - `test-measure.sh exit=0`
  - `test-mogui-logging.sh exit=0`
  - `test-product-path-guard.sh exit=0`
  - `test-seat-check.sh exit=0`
  - `test-spawn-test.sh exit=0`
  - `test-tool-naming.sh exit=0`
  - `test-tracker-check-banner.sh exit=0`
  - `test-worker-block-warn.sh exit=0`
  - `test-worker-pane-sweep.sh exit=0`
  - `passed 12 failed 0 skip-lines 0`
- [x] `git diff origin/main...HEAD | grep '^-'` and removal rationale:
  - `--- a/.github/workflows/gates.yml` (diff header line only; no workflow behavior was removed)

## Review

I replied to each active review-bot thread. I accepted and fixed the empty-match edge case in commit `0a72ca7`, and documented why I kept the contract-required two-step shell-tests job shape for the Python setup suggestion.

## Changelog

none

## Template impact

none

## Notes

A clean `git archive` without `.git` is still known to break `test-spawn-test.sh` because `spawn-test` calls `git rev-parse` under `set -e` while sourced; this remains unchanged and was only reported per contract scope.

### CODE REVIEW

File: master-ops/scripts/test-zz-probe.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "probe: intentional failure for shell-tests CI gate"
+exit 1
```
[the remark this answers is not in the document]
Comment (baksohyeon) [reply master-ops/scripts/test-zz-probe.sh]: Confirmed. The temporary probe was removed in commit c1b2a40, and shell-tests returned to success on both legs in run https://github.com/baksohyeon/mogui-ADE-orchestrator/actions/runs/35335258076.

File: .github/workflows/gates.yml [hunk 2]
```diff
@@ -72,6 +72,48 @@ jobs:
...
+    # No Windows leg: these tests require the Unix script execution surface.
+    # Probe coverage for the failure path was proven in CI on 2026-09-18.
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v7.0.1
+      - name: Run promoted shell tests
```
[the remark this answers is not in the document]
Comment (baksohyeon) [reply .github/workflows/gates.yml:89]: Good catch on Python availability risk. I kept the shell-tests job to the contract shape (checkout plus one bash step), and verified the current branch run https://github.com/baksohyeon/mogui-ADE-orchestrator/actions/runs/35335785250 where both shell-tests legs completed successfully with this structure.

File: .github/workflows/gates.yml [hunk 3]
```diff
@@ -72,6 +72,49 @@ jobs:
...
+        shell: bash
+        run: |
+          set -u
+          shopt -s nullglob
```
[the remark this answers is not in the document]
Comment (baksohyeon) [reply .github/workflows/gates.yml:93]: Confirmed and fixed. I now collect matches into an array, fail explicitly when the set is empty, and sort the populated array before execution in commit 0a72ca7.

File: .github/workflows/gates.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (baksohyeon) [reply .github/workflows/gates.yml:89]: The shell-tests job now provisions Python 3.12 with actions/setup-python@v7.0.0 before running the promoted shell tests, which removes runner-image Python variance. Fixed in commit d408cc3.
